internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/servers"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/virtualendpoints"
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres/custompollers"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type PostgresqlFlexibleServerVirtualEndpointResource struct{}
+
+type PostgresqlFlexibleServerVirtualEndpointModel struct {
+	Name            string `tfschema:"name"`
+	SourceServerId  string `tfschema:"source_server_id"`
+	ReplicaServerId string `tfschema:"replica_server_id"`
+	Type            string `tfschema:"type"`
+}
+
+var _ sdk.ResourceWithUpdate = PostgresqlFlexibleServerVirtualEndpointResource{}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) ModelObject() interface{} {
+	return &PostgresqlFlexibleServerVirtualEndpointModel{}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) ResourceType() string {
+	return "azurerm_postgresql_flexible_server_virtual_endpoint"
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return virtualendpoints.ValidateVirtualEndpointID
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The name of the Virtual Endpoint",
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"source_server_id": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The Resource ID of the *Source* Postgres Flexible Server this should be associated with",
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: servers.ValidateFlexibleServerID,
+		},
+		"replica_server_id": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The Resource ID of the *Replica* Postgres Flexible Server this should be associated with",
+			Required:     true,
+			ValidateFunc: servers.ValidateFlexibleServerID,
+		},
+		"type": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The type of Virtual Endpoint",
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(virtualendpoints.PossibleValuesForVirtualEndpointType(), true),
+		},
+	}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
```

Comment:
We may want to bump this to 10
```suggestion
		Timeout: 10 * time.Minute,
```

```
        Error: creating Virtual Endpoint (Subscription: "*******"
        Resource Group Name: "acctest-ve-rg-240808090324667981"
        Flexible Server Name: "acctest-ve-primary-240808090324667981"
        Virtual Endpoint Name: "acctest-ve-240808090324667981"): polling after Create: context deadline exceeded
```

Reply:
changed